sonar-helm-for-iac/converters/io-handling.go

```diff
@@ -39,37 +41,56 @@ type InputReader interface {
 	// <template content>
 	// ...
 	// END
-	ReadInput(scanner *bufio.Scanner) []Content
+	ReadInput(scanner *bufio.Scanner) ([]Content, error)
 }
 
 type StdinReader struct{}
 
-func (s StdinReader) ReadInput(scanner *bufio.Scanner) []Content {
+func (s StdinReader) ReadInput(scanner *bufio.Scanner) ([]Content, error) {
 	contents := make([]Content, 0)
-	firstLine := s.readInput(scanner, 1)
+	firstLine, err := s.readInput(scanner, 1)
+	if err != nil {
+		return nil, fmt.Errorf("read input: %w", err)
+	}
 	if firstLine == "" {
 		fmt.Fprintf(os.Stderr, "Received empty input, exiting\n")
-		return contents
+		return contents, nil
 	}
+
 	for firstLine != "END" {
 		name := firstLine
-		length, _ := strconv.Atoi(s.readInput(scanner, 1))
-		fmt.Fprintf(os.Stderr, "Reading %d lines from stdin\n", length)
-		content := s.readInput(scanner, length)
+
+		var content string
+		contentResult := mo.TupleToResult(s.readInput(scanner, 1)).FlatMap(
```

Comment:
So the idea here is to read 3 different things from stdin and handle errors. Handling errors will be the same all the time: abort reading, return an error upward. However, with vanilla Go it requires too much boilerplate. Using `mo.Result` allows to declare set of operations, and `err` will be set once first failure occurs; all other operations won't be executed.
The library is not perfect, e.g. doesn't allow mapping between different types, but may come in handy